bus-mapping/src/circuit_input_builder/block.rs

```diff
@@ -217,6 +225,14 @@ pub struct Block {
     /// relax mode indicate builder and circuit would skip
     /// some sanity check, used by testing and debugging
     relax_mode: bool,
+    /// Previous last applied l1 block
+    pub prev_last_applied_l1_block: Option<u64>,
+    /// Last applied l1 block
+    pub last_applied_l1_block: Option<u64>,
+    /// L1 block hashes
+    pub cum_l1_block_hashes: Vec<Hash>,
+    /// L1 block range hash
+    pub l1_block_range_hash: Option<Hash>,
```

Comment:
I think you can remove this field? This can be computed from the above field, i.e. `cum_l1_block_hashes`. So it would make sense to remove this and instead add a utility method to the `Block` struct. Something like:
```rust
impl Block {
    fn l1_block_range_hash(&self) -> Option<Hash> {
        /* compute from self.cum_l1_block_hashes */
    }
}
```

Reply:
the `l1_block_range_hash` field comes from the prover and/or verifier as public input so we can compare it to the keccak, i.e. `l1_block_range_hash == keccak(l1_block_hashes[0], l1_block_hashes[1], ...)`